src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -492,6 +558,29 @@ std::optional<DataModel::AttributeInfo> CodegenDataModelProvider::GetAttributeIn
 
 DataModel::CommandEntry CodegenDataModelProvider::FirstAcceptedCommand(const ConcreteClusterPath & path)
 {
+    CommandHandlerInterface * interface =
+        CommandHandlerInterfaceRegistry::Instance().GetCommandHandler(path.mEndpointId, path.mClusterId);
+
+    if (interface != nullptr)
+    {
+        EnumeratorCommandFinder finder(EnumeratorCommandFinder::Operation::FindFirst, kInvalidCommandId);
+        CHIP_ERROR err = interface->EnumerateAcceptedCommands(path, finder.Callback(), &finder);
+
+        if (err != CHIP_ERROR_NOT_IMPLEMENTED)
+        {
+            auto firstId = finder.GetFound();
+
+            if ((err == CHIP_NO_ERROR) && firstId.has_value())
+            {
+                DataModel::CommandEntry result;
+                result.path = ConcreteCommandPath(path.mEndpointId, path.mClusterId, *firstId);
```

Comment:
So this is just silently not setting up state that the caller expects?

In practice the way this stuff works right now is that `EnumerateAcceptedCommands` is expected to return a _subset_ of what's enabled in the ZAP file, so that we have the right timed/etc metadata in the places that use it.  If someone hase `EnumerateAcceptedCommands` return commands that are not in the ZAP config, I guess the existing code runs and does the wrong things.   But this new code will run and do the wrong things in the case when the API is actually implemented correctly, no?

It seems to me like we should be calling `CommandEntryFrom` here, not manually constructing a CommandEntry.

Reply:
Ok, I can make it that "if zap also contains this, return its info". I was not aware that we enforce always returning a subset (although that would absolutely make it easier to have correct flags)

Reply:
Updated to use the right constructor to forward the info.

Reply:
> I was not aware that we enforce always returning a subset

We don't, but in practice all existing use cases of this do that, and if you don't you will be violating the spec in various ways...